packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -123,66 +124,178 @@ contract L2Genesis is Script, Artifacts {
             address(prefix | uint160(0)),
-            address(prefix | uint160(PROXY_COUNT - 1))
+            address(prefix | uint160(PREDEPLOY_COUNT - 1))
         );
-        for (uint256 i = 0; i < PROXY_COUNT; i++) {
+        for (uint256 i = 0; i < PREDEPLOY_COUNT; i++) {
             address addr = address(prefix | uint160(i));
             if (_notProxied(addr)) {
+                console.log("Skipping proxy at %s", addr);
                 continue;
             }
 
             vm.etch(addr, code);
-            vm.store(addr, PROXY_ADMIN_ADDRESS, bytes32(uint256(uint160(Predeploys.PROXY_ADMIN))));
+            EIP1967Helper.setAdmin(addr, Predeploys.PROXY_ADMIN);
 
             if (_isDefinedPredeploy(addr)) {
-                address implementation = _predeployToCodeNamespace(addr);
+                address implementation = predeployToCodeNamespace(addr);
                 console.log("Setting proxy %s implementation: %s", addr, implementation);
-                vm.store(addr, PROXY_IMPLEMENTATION_ADDRESS, bytes32(uint256(uint160(implementation))));
+                EIP1967Helper.setImplementation(addr, implementation);
             }
         }
     }
 
-    /// @notice LEGACY_ERC20_ETH is not being predeployed since it's been deprecated.
     /// @dev Sets all the implementations for the predeploy proxies. For contracts without proxies,
     ///      sets the deployed bytecode at their expected predeploy address.
-    function _setPredeployImplementations() internal {
-        _setLegacyMessagePasser();
-        _setDeployerWhitelist();
-        _setWETH9();
-        _setL2StandardBridge();
-        _setL2CrossDomainMessenger();
-        _setSequencerFeeVault();
-        _setOptimismMintableERC20Factory();
-        _setL1BlockNumber();
-        _setGasPriceOracle();
-        _setGovernanceToken();
-        _setL1Block();
+    ///      LEGACY_ERC20_ETH and L1_MESSAGE_SENDER are deprecated and are not set.
+    function setPredeployImplementations() internal {
+        // TODO: sort by address, is this order dependent?
+        setProxyAdmin();
+        setL2ToL1MessagePasser();
+        setL2CrossDomainMessenger();
+        setL2StandardBridge();
+        setL2ERC721Bridge();
+        setSequencerFeeVault();
+        setOptimismMintableERC20Factory();
+        setOptimismMintableERC721Factory();
+        setL1Block();
+        setGasPriceOracle();
+        setDeployerWhitelist();
+        setWETH9();
+        setL1BlockNumber();
+        setLegacyMessagePasser();
+        setBaseFeeVault();
+        setL1FeeVault();
+        setGovernanceToken();
+        setSchemaRegistry();
+        setEAS();
+        setProxyAdmin();
+    }
+
+    function setProxyAdmin() public {
+        // Note the ProxyAdmin implementation itself is behind a proxy that owns itself.
+        _setImplementationCode(Predeploys.PROXY_ADMIN);
+
+        bytes32 _ownerSlot = hex"0000000000000000000000000000000000000000000000000000000000000000";
```

Comment:
You can represent this as `bytes32(0)` more cleanly

Reply:
I thought about it, but this is consistent with all the other slot declarations. I'll change it to `bytes32(0)` if you like that better though.

Reply:
I think it would be much cleaner to do `bytes32(0)`, less characters, easier to update + verify correctness, the old stuff was just there and should be updated to new syntax if necessary as well